components/reports/QueueFilter/SubjectType.tsx

```diff
@@ -20,6 +20,40 @@ export const QueueFilterSubjectType = () => {
       return tag.startsWith('embed:')
     }) || []
 
+  return (
+    <SubjectTypeFilter
+      hasSubjectTypeFilter={hasSubjectTypeFilter}
+      isSubjectTypeAccount={queueFilters.subjectType === 'account'}
+      selectedCollections={selectedCollections}
+      isSubjectTypeRecord={
+        queueFilters.subjectType === 'record' ||
+        !!selectedCollections.length ||
+        !!selectedIncludeEmbedTypes.length
+      }
+      toggleCollection={toggleCollection}
+      toggleSubjectType={toggleSubjectType}
+      clearSubjectType={clearSubjectType}
+    />
+  )
+}
+
+export const SubjectTypeFilter = ({
```

Comment:
Refactored the UI part of it into its own component so that we can reuse this in queue filter and event list filter.